src/helpers/serverModuleHelpers.js

```diff
@@ -0,0 +1,187 @@
+  if (!satisfies(angularVersion, '>=19.0.0-rc', { includePrerelease: true })) {
+    // for pre-19 versions, we don't need to do anything
+    return
+  }
+
+  const angularJson = getAngularJson({ failPlugin, siteRoot })
+
+  const project = getProject(angularJson)
+  const {
+    architect: { build },
+  } = project
+
+  serverModuleLocation = build?.options?.ssr?.entry
+  if (!serverModuleLocation || !existsSync(serverModuleLocation)) {
+    return
+  }
+
+  // check wether project is using stable CommonEngine or Developer Preview AppEngine
+  const serverModuleContents = await readFile(serverModuleLocation, 'utf8')
+
+  const usedEngineBasedOnKnownSignatures = getEngineBasedOnKnownSignatures(serverModuleContents)
+  if (usedEngineBasedOnKnownSignatures) {
+    needSwapping = true
+
+    console.log(
+      `Default server.ts using ${usedEngineBasedOnKnownSignatures} found. Automatically swapping to Netlify compatible server.ts.`,
+    )
+
+    const parsed = parse(serverModuleLocation)
+
+    serverModuleBackupLocation = join(parsed.dir, `${parsed.name}.original${parsed.ext}`)
+
+    await rename(serverModuleLocation, serverModuleBackupLocation)
+
+    if (usedEngineBasedOnKnownSignatures === 'CommonEngine') {
+      await writeFile(serverModuleLocation, NetlifyServerTsCommonEngine)
+    } else if (usedEngineBasedOnKnownSignatures === 'AppEngine') {
+      await writeFile(serverModuleLocation, NetlifyServerTsAppEngine)
+    }
+    return usedEngineBasedOnKnownSignatures
+  }
+
+  // if we can't determine engine based on known signatures, let's first try to check if module is already
+  // Netlify compatible to determine if it can be used as is or if user intervention is required
+  // we will look for "netlify<Engine>Handler" which is named export that we will rely on and it's existence will
+  // be quite strong indicator that module is already compatible and doesn't require any changes
+
+  const isNetlifyAppEngine = serverModuleContents.includes('netlifyAppEngineHandler')
+  const isNetlifyCommonEngine = serverModuleContents.includes('netlifyCommonEngineHandler')
+
+  if (isNetlifyAppEngine && isNetlifyCommonEngine) {
+    // both exports found - we can't determine which engine is used
+    failBuild(
+      "server.ts seems to contain both 'netlifyAppEngineHandler' and 'netlifyCommonEngineHandler' - it should contain just one of those.",
+    )
+  }
+
+  if (isNetlifyAppEngine) {
+    return 'AppEngine'
+  }
+
+  if (isNetlifyCommonEngine) {
+    return 'CommonEngine'
+  }
+
+  // at this point we know that user's server.ts is not Netlify compatible so user intervention is required
+  // we will try to inspect server.ts to determine which engine is used and provide more accurate error message
+  const guessedUsedEngine = guessUsedEngine(serverModuleContents)
+
+  let errorMessage = `server.ts doesn't seem to be Netlify compatible and is not known default. Please replace it with Netlify compatible server.ts.`
+  if (guessedUsedEngine) {
```

Comment:
this error case can be seen in https://app.netlify.com/sites/plugin-angular-universal-demo/deploys/673b7bee10573500083ea36f